### ISSUE TITLE
super-with-arguments should not be shown for dataclasses with slots

### ISSUE THREAD

**Issue by Infernio** [PyCQA/pylint#5823]
### Bug description

Python 3.10 adds `slots=True` to dataclass decorators. The documentation describes it as such (emphasis mine):
> slots: If true (the default is False), `__slots__` attribute will be generated and **new class will be returned instead of the original one**. If `__slots__` is already defined in the class, then TypeError is raised.

The fact that it returns a new class breaks argument-less super, as shown in this example:

```python
"""Quick test showing that dataclasses with slots break argument-less super()
and hence pylint should not suggest using it."""
from dataclasses import dataclass

@dataclass(slots=True)
class _Animal:
    """Base class for animals."""
    name: str
    num_arms: int
    num_legs: int

    def greet(self):
        """Simple method to use all fields in this dataclass."""
        print(f'Hi, my name is {self.name} and I have {self.num_arms} arms '
              f'and {self.num_legs} legs.')

@dataclass(slots=True)
class Dog(_Animal):
    """Dataclass implementing a dog."""
    num_arms: int = 0
    num_legs: int = 4

    def greet(self):
        print('Woof!')
        # This works
        super(Dog, self).greet()

@dataclass(slots=True)
class Bird(_Animal):
    """Dataclass implementing a bird."""
    num_arms: int = 0
    num_legs: int = 2

    def greet(self):
        print('Chirp!')
        # This does *not* work
        super().greet()

Dog('Snoopy').greet()
# This will raise a TypeError, see Bird.greet above
Bird('Blu').greet()
```

`Bird` implements pylint's suggestion to use argument-less super,  `Dog` does not. Running the file:
```
$ python dc_slots.py
Woof!
Hi, my name is Snoopy and I have 0 arms and 4 legs.
Chirp!
Traceback (most recent call last):
  File "/tmp/dc_slots.py", line 40, in <module>
    Bird('Blu').greet()
  File "/tmp/dc_slots.py", line 37, in greet
    super().greet()
TypeError: super(type, obj): obj must be an instance or subtype of type
```

### Command used

```shell
python -m pylint dc_slots.py
```

### Pylint output

```shell
************* Module dc_slots
dc_slots.py:26:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

------------------------------------------------------------------
Your code has been rated at 9.52/10 (previous run: 9.52/10, +0.00)
```

### Expected behavior

pylint should not be suggesting an argument-less super call in dataclasses that have `slots=True`.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.2 (main, Jan 15 2022, 19:56:27) [GCC 11.1.0]
```

### OS / Environment

Arch Linux
